tools/yapf.sh

```diff
@@ -36,14 +36,20 @@ function follow_links() (
 SCRIPT_DIR=$(follow_links "$(dirname -- "${BASH_SOURCE[0]}")")
 SRC_DIR="$(cd "$SCRIPT_DIR/../.."; pwd -P)"
 YAPF_DIR="$(cd "$SRC_DIR/flutter/third_party/yapf"; pwd -P)"
+
+# TODO: https://github.com/flutter/flutter/issues/158384
+# Migrate to a supported Python formatter.
 if command -v python3.10 &> /dev/null; then
   PYTHON_EXEC="python3.10"
 else
   python3 -c "
 import sys
 version = sys.version_info
 if (version.major, version.minor) > (3, 11):
-    print(f'Error: python3 version {version.major}.{version.minor} is greater than 3.11.', file=sys.stderr)
+    print(f'Error: The yapf Python formatter requires Python version 3.11 or '
+          f'earlier. The installed python3 version is '
```

Comment:
The script selects 3.10 and yapf only advertises supporting up to 3.10.  Should we just talk about 3.10 (despite we allowed 3.11 in for our linux servers)?  Maybe we could expand the selection of PYTHON_EXEC to include 3.11?

Reply:
If we know it works at 3.11 on the linux servers, seems fair to give people a little more leeway. I've expanded the selection to prefer 3.10 but fall back (forward?) to 3.11 if 3.10 isn't available but 3.11 is.